### DISCUSSION TITLE
Функция linkToTable

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#224] [Как это сделать?] [answered]
Как передать в linkToTable значение из текущей таблицы, строки, поля в фильтр?

**Comment by ttm-alexeypolunin** [comment]
Из текущей таблицы и текущей строки значения беруться через `#field_name` https://ru.docs.totum.online/calculation-of-the-value-by-code

В linkToTable они передаются в параметр `filter: 'fl_prefilter_name' = #field_name_with_value`: https://ru.docs.totum.online/how-do-I-open-another-table-at-the-click-of-a-button

Что бы поле префильтра заполнилось таким образом, оно должно быть доступно для редактирования для открывающей роли.

**Comment by vldekov** [comment, marked as answer]
написал так =:linkToTable(table: 'analiz'; title: $fix2; filter: 'fl_nomersl' = #nomersl), но выходит ошибка при выполнении "Поле nomersl не найдено. Функция linkToTable; action поле h_knopkaanaliz таблицы tablicasl"

**Reply by erpic-ru** [reply]
Я такую кнопку обычно ставлю в панель по правой кнопке в код форматирования ячейки: 
```
p1=:panelButtons(button: $1; )

1: rowCreate(field: "text" = "Фильтр"; field: "code" = $kot; field: "icon" = "cat";field: "background" = "lightgreen";)

kot: strAdd(str: '=: linkToTable(table: $#ntn; filter: "fl_voditel" = $#nfv; target: "self";  refresh: false;)')
```

**Reply by vldekov** [reply]
Порядок поля кнопки равен 50, а поля `#nomersl` равен 8. Поле кнопки находится в хедере, а поле `#nomersl` в строчной части. Что здесь не так?

Вот такой код работает:

```
=:linkToTable(table: 'analiz'; title: $fix2; filter: 'fl_nomersl' = $fix1; target: "iframe"; width: 1500; refresh: true; header: true; footer: true)
fix1: select(table: 'tablicasl'; field: 'nomersl';where: 'id' = $#ids)
fix2: str`"Все рекомендации по СЛ № " + $fix1`
```
Но это менее удобно чем просто когда пользователь встал на строчку, поэтому хотелось бы не требовать от него отметить ID строчки.

**Reply by erpic-ru** [reply]
Можно сделать как в коде выше, если только по одному значению отфильтровать надо или отдельное поле с кнопкой в табличной части разместить. 

Либо как-то иначе эту задачу решать

**Reply by erpic-ru** [reply]
Саму кнопку можно сделать узкую без названия с одной иконкой. Тогда обращение будет #nomersl без всяких селектов и лишних кликов.
<img width="288" alt="Снимок экрана 2022-09-26 в 15 31 59" src="https://user-images.githubusercontent.com/91674114/192277123-b69b3496-fb15-42e9-9142-775f78e832e8.png">

**Reply by ttm-alexeypolunin** [reply]
Порядок полей по sort  не влияет на доступ к действиям. Дело в том, что `h_knopkaanaliz` находится в хедере, а `nomersl` в строчной части.

**Пока** выделение строки курсором никак нельзя передать в выполняемый код действия.

Поэтому:

— использовать выделение галочкой
— контекстное меню
— кнопка в строке

Мы подумаем над тем, что бы была быстрая переменная, которая будет знать, какие строки выделены курсором, но это точно в PRO.

**Reply by vldekov** [reply]
Взамен кнопок в хедере выбрал вариант linkToButtons. Возможно настроить эту функцию, чтобы после нажатия в ней кнопок эта панель с кнопками сама не закрывалась?

**Reply by ttm-alexeypolunin** [reply]
Базово нет никакого параметра, который бы позволял удержать окно. Но можно немного изнасиловать систему и сделать код кнопки, который откроет новый такой же `linkToButtons`:

```
= : linkToButtons(title: "test"; buttons: $row_list; refresh: true)

row_list: listCreate(item: $row)
	row: rowCreate(field: "text" = "SOME BUTTON"; field: "code" = $code; field: "vars" = $vars)
		vars: rowCreate(field: "var_c" = $code)
	
    ```code:totum
    a1=: linkToDataText(title: "YOUR MAIN CODE IN THIS LINE"; text: "MAIN CODE")
    
    a2=: linkToButtons(title: "test"; buttons: $row_list; refresh: true)
    
    row_list: listCreate(item: $row)
    	row: rowCreate(field: "text" = "SOME BUTTON"; field: "code" = $#var_c; field: "vars" = $vars)	
		vars: rowCreate(field: "var_c" = $#var_c)
    ```

```

**Reply by ttm-alexeypolunin** [reply]
> Мы подумаем над тем, что бы была быстрая переменная, которая будет знать, какие строки выделены курсором, но это точно в PRO.

**$#slPro** — только в PRO. Переменная доступна только в коде действия Кнопки аналогично `$#ids`. получает выделенные мышкой поля аналогично структуре `$#changes`.

**Comment by vldekov** [comment]
Спасибо за хорошие новости.Подскажите порядок обновления с версии 3.7 до версии 4.8?

**Reply by ttm-alexeypolunin** [reply]
```
bin/totum git-update --force
```
